src/compiler/tsbuildPublic.ts

```diff
@@ -962,35 +983,31 @@ namespace ts {
             // Actual Emit
             const { host, compilerHost } = state;
             let resultFlags = BuildResultFlags.DeclarationOutputUnchanged;
-            let newestDeclarationFileContentChangedTime = minimumDate;
-            let anyDtsChanged = false;
+            const existingBuildInfo = state.buildInfoCache.get(projectPath)?.buildInfo || undefined;
             const emitterDiagnostics = createDiagnosticCollection();
             const emittedOutputs = new Map<Path, string>();
-            outputFiles.forEach(({ name, text, writeByteOrderMark }) => {
-                let priorChangeTime: Date | undefined;
-                if (!anyDtsChanged && isDeclarationFileName(name)) {
-                    // Check for unchanged .d.ts files
-                    if (host.fileExists(name) && state.readFileWithCache(name) === text) {
-                        priorChangeTime = host.getModifiedTime(name);
-                    }
-                    else {
+            const options = program.getCompilerOptions();
+            const isIncremental = isIncrementalCompilation(options);
+            let outputTimeStampMap: ESMap<Path, Date> | undefined;
+            let now: Date | undefined;
+            outputFiles.forEach(({ name, text, writeByteOrderMark, buildInfo }) => {

```

Comment:
Here earlier we use to save if d.ts emit was done by reading file on disk.. Now we just use time stamp from buildInfo..
Also we are saving buildInfo in cache so we dont need to read it again from the disk